pkg/cmd/auth/login/login.go

```diff
@@ -51,13 +54,24 @@ func NewCmdLogin(f *cmdutil.Factory, runF func(*LoginOptions) error) *cobra.Comm
 			$ gh auth login
 			# => do an interactive setup
 
+			$ gh auth login --web
+			# => open a browser to authenticate and do a non-interactive setup
```

Comment:
I'm not sure if this mode of setup can be called non-interactive, since it requires keypress interaction and going through a graphical web browser.